apps/emqx_authn/src/simple_authn/emqx_authn_http.erl

```diff
@@ -261,21 +263,47 @@ transform_header_name(Headers) ->
     ).
 
 check_ssl_opts(Conf) ->
-    {BaseUrl, _Path, _Query} = parse_url(get_conf_val("url", Conf)),
-    case BaseUrl of
-        <<"https://", _/binary>> ->
-            case get_conf_val("ssl.enable", Conf) of
-                true -> ok;
-                false -> false
+    case is_backend_http(Conf) of
+        true ->
+            Url = get_conf_val("url", Conf),
+            {BaseUrl, _Path, _Query} = parse_url(Url),
+            case BaseUrl of
+                <<"https://", _/binary>> ->
+                    case get_conf_val("ssl.enable", Conf) of
+                        true ->
+                            ok;
+                        false ->
+                            <<"it's required to enable the TLS option to establish a https connection">>
```

Comment:
we can simply use a converter to enable ssl ?

Reply:
If we help user to enabled ssl, how to configure `#{enable => true, cacertfile => ... certfile => ....keyfile => ...}`
I thinks it must provide by user.

Reply:
the default is verify_none, so there is usually no need to configure cacerts.
and typically http servers do not validate client certificates, so the cert files typically not needed.